src/test/java/school/hei/haapi/integration/TeacherIT.java

```diff
@@ -138,11 +157,76 @@ void teacher_read_ko() {
 
     UsersApi api = new UsersApi(teacher1Client);
     assertThrowsApiException(
-        "{\"type\":\"403 FORBIDDEN\",\"message\":\"Access is denied\"}",
-        () -> api.getTeacherById(TEACHER2_ID));
+            "{\"type\":\"403 FORBIDDEN\",\"message\":\"Access is denied\"}",
+            () -> api.getTeacherById(TEACHER2_ID));
     assertThrowsApiException(
-        "{\"type\":\"403 FORBIDDEN\",\"message\":\"Access is denied\"}",
-        () -> api.getTeachers(1, 20));
+            "{\"type\":\"403 FORBIDDEN\",\"message\":\"Access is denied\"}",
+            () -> api.getTeachers(1, 20, null, null, null));
+  }
+
+  @Test
```

Comment:
No. Use intergration test for just teacher for example, but use unit tests for the rest.

Reply:
After thinking a while, is it really necessary to add UT for teacher controller if i just added default values into @RequestParams and the service, already tested in StudentIT, is doing the rest.